### PR TITLE
Add util function for pretty printing of output diffs

### PR DESCRIPTION
**Pull request by AdrianLundell**
The function is added as a callback to run_method_and_compare_outputs to open up for other types  of analyze methods in the future.

### CODE REVIEW

**Comment by AdrianLundell** [comment]
Unrelated CI failure in 
- backends/xnnpack/test/ops/test_add.py::TestAdd::test_qs8_add:  https://github.com/pytorch/executorch/actions/runs/12300250961/job/34327986575?pr=7302#step:14:7721
- examples/models/llama3_2_vision/text_decoder/test/test_text_decoder.py: https://github.com/pytorch/executorch/actions/runs/12300250961/job/34327988517?pr=7302#step:9:16682

File: backends/arm/test/tester/analyze_output_utils.py [hunk 1]
```diff
@@ -0,0 +1,216 @@
+# Copyright 2024 Arm Limited and/or its affiliates.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+import logging
+
+import torch
+
+logger = logging.getLogger(__name__)
```
Comment (freddan80) [inline backends/arm/test/tester/analyze_output_utils.py:10]:
```suggestion
logger = logging.getLogger(__name__) # Test suggest change
```

Comment (freddan80) [reply backends/arm/test/tester/analyze_output_utils.py:10]: Ignore. Just a test

**Comment by AdrianLundell** [comment]
Unrelated CI error when running 

```
curl -Lo /tmp/v2.25.0.24.07.28.zip https://softwarecenter.qualcomm.com/api/download/software/qualcomm_neural_processing_sdk/v2.25.0.240728.zip 
...
2024-12-18T08:56:22.4369579Z curl: (92) HTTP/2 stream 0 was not closed cleanly: INTERNAL_ERROR (err 2)
2024-12-18T08:56:22.4370216Z     raise RuntimeError(f"Command {cmd} failed with exit code {exit_code}")
```

**Review by freddan80 — APPROVED** [review]

**Pull request merged**